src/Microsoft.OData.Core/Json/IJsonWriterFactory.cs

```diff
@@ -18,9 +19,10 @@ public interface IJsonWriterFactory
         /// <summary>
         /// Creates a new JSON writer of <see cref="IJsonWriter"/>.
         /// </summary>
-        /// <param name="textWriter">Writer to which text needs to be written.</param>
+        /// <param name="stream">The stream to write to.</param>
         /// <param name="isIeee754Compatible">True if it is IEEE754Compatible.</param>
+        /// <param name="encoding">The character encoding to use.</param>
         /// <returns>The JSON writer created.</returns>
-        IJsonWriter CreateJsonWriter(TextWriter textWriter, bool isIeee754Compatible);
+        IJsonWriter CreateJsonWriter(Stream stream, bool isIeee754Compatible, Encoding encoding);

```

Comment:
Why not use StreamWriter ?

Reply:
@wachugamaina Why `StreamWriter`?

Reply:
@wachugamaina Passing the output stream allows the freedom for the `IJsonWriter` implementation to choose which means to use to write to the stream, e.g., write to the `Stream` directly by calling the `Write`/`WriteAsync` methods, use `StreamWriter` like you suggested, use `Utf8JsonWriter`, etc.